packages/embeds/js/src/components/InputChatBlock.tsx

```diff
@@ -93,10 +93,11 @@ export const InputChatBlock = (props: Props) => {
     const formattedMessage = formattedMessages().findLast(
       (message) => props.chunkIndex === message.inputIndex,
     )?.formattedMessage;
+
     if (formattedMessage && props.block.type !== InputBlockType.FILE)
       setAnswer((answer) =>
         answer?.type === "text"
-          ? { ...answer, label: formattedMessage }
+          ? { ...answer, label: answer.label ?? formattedMessage }
           : answer,
       );
   });
```

Comment:
This can indeed be generalized, if the input has a label applied, we shouldn't overwrite it

```suggestion
    if (answer.label) return
    const formattedMessage = formattedMessages().findLast(
      (message) => props.chunkIndex === message.inputIndex,
    )?.formattedMessage;

    if (formattedMessage)
      setAnswer((answer) =>
        answer?.type === "text"
          ? { ...answer, label: answer.label ?? formattedMessage }
          : answer,
      );
  });
```